### DISCUSSION TITLE
Real Time Error and subscription is not seeing payload

### DISCUSSION THREAD

**Discussion by Iroh-Omolola** [supabase/supabase#13178] [Questions]
RealtimeChannel {topic: 'realtime:public:messages', params: {…}, socket: RealtimeClient, bindings: {…}, state: 'errored', …}
bindings
: 
{phx_close: Array(2), phx_error: Array(2), phx_reply: Array(1), presence_state: Array(1), presence_diff: Array(1), …}
joinPush
: 
Push {channel: RealtimeChannel, event: 'phx_join', payload: {…}, timeout: 10000, sent: true, …}
joinedOnce
: 
true
params
: 
{config: {…}}
presence
: 
RealtimePresence {channel: RealtimeChannel, state: {…}, pendingDiffs: Array(0), joinRef: null, caller: {…}}
pushBuffer
: 
[]
rejoinTimer
: 
Timer {timer: 293, tries: 66, callback: ƒ, timerCalc: ƒ}
socket : RealtimeClient {accessToken: 'eyJhbGciOi................, channels: Array(1), endPoint: 'wss://ryxsuocomobnsuiyevpv.supabase.co/realtime/v1/websocket', headers: {…}, params: {…}, …}
state : "errored"
timeout : 10000
topic : "realtime:public:messages"

 messageSubscription = supabase
    .channel("public:messages")
    .on(
      "postgres_changes",
      { event: "INSERT", schema: "public", table: "messages" },
      (payload) => {
        console.log("Change received!", payload);
        setMessages((messages) => [...messages, payload.new]);
      }
    )
    .subscribe((change) => console.log("todos changed", change));

    return () => {
      supabase.removeAllChannels()
    }

I am working from local

**Comment by w3b6x9** [comment]
@Iroh-Omolola did you follow the docs here? https://supabase.com/docs/guides/realtime/extensions/postgres-changes

**Reply by Iroh-Omolola** [reply]
yes. 
am I missing out something?

@w3b6x9  does it work on local?
I have done the needful, still getting "status:errored" on real time

Here is the sql:

create table messages (
  id uuid primary key default gen_random_uuid(),
  message text not null,
  username text not null,
  room text not null,
  created_at timestamp with time zone default timezone('utc'::text, now()) not null
);


begin;
  -- remove the realtime publication
  drop publication if exists supabase_realtime;

  -- re-create the publication but don't enable it for any tables
  create publication supabase_realtime;
commit;


alter table
  messages replica identity full;

**Reply by w3b6x9** [reply]
you need to add the table to the publication:

```sql
alter publication supabase_realtime add table messages;
```

**Reply by w3b6x9** [reply]
it's the last step in this section: https://supabase.com/docs/guides/realtime/extensions/postgres-changes#replication-setup

**Reply by Iroh-Omolola** [reply]
@w3b6x9 yes, I did, still same error.
I get these error first from the subscription:
CHANNEL_ERROR
CLOSED

And this:
RealtimeChannel {topic: 'realtime:public:messages', params: {…}, socket: RealtimeClient, bindings: {…}, state: 'errored', …}
bindings
: 
{phx_close: Array(2), phx_error: Array(2), phx_reply: Array(1), presence_state: Array(1), presence_diff: Array(1), …}
joinPush
: 
Push {channel: RealtimeChannel, event: 'phx_join', payload: {…}, timeout: 10000, sent: true, …}
joinedOnce
: 
true
params
: 
{config: {…}}
presence
: 
RealtimePresence {channel: RealtimeChannel, state: {…}, pendingDiffs: Array(0), joinRef: null, caller: {…}}
pushBuffer
: 
[]
rejoinTimer
: 
Timer {timer: 89, tries: 15, callback: ƒ, timerCalc: ƒ}
socket
: 
RealtimeClient {accessToken: {access}, channels: Array(1), endPoint: 'wss://ryxsuocomobnsuiyevpv.supabase.co/realtime/v1/websocket', headers: {…}, params: {…}, …}
state
: 
"errored"
timeout
: 
10000
topic
: 
"realtime:public:messages"

**Reply by Iroh-Omolola** [reply]
@w3b6x9 does this work on local?

**Reply by w3b6x9** [reply]
can you submit a support ticket? i think this is specific to your supabase project and/or your local setup.

**Reply by Iroh-Omolola** [reply]
okay. thanks

**Reply by Iroh-Omolola** [reply]
@w3b6x9 

this doesn't work
 async function signInWithEmail() {
  const { data, error } = await supabase.auth.signInWithPassword({
    email: 'example@email.com',
    password: 'example-password',
  })
} 

but sign up does.
